### PR TITLE
fix #3327 remove and ban hardcoded strings

### LINKED ISSUE #3327: Remove all hardcoded strings from the app
We should remove all hardcoded strings from the app:

* Some should be replaced by translatable strings (e.g. `Start Date` in `res/layout/date_range_dialog.xml`), 
* some by non-translatable strings (e.g. `automattic.com` in `res/layout/about_activity.xml`), 
* some can be dropped (e.g. `This is a text` in `res/layout/theme_details_fragment.xml`) 
* and some can be replaced with `tools:text` if needed in the visual editor (e.g. `date` in `res/layout/comment_listitem.xml`).

Run the following command to list them: `pushd WordPress/src/main ; lint --check HardcodedText . ; popd`

Once this is done, we must define the `HardcodedText` lint check as an error instead of a warning.

The reason behind that: make sure we can't add new hardcoded strings by mistake.

### PR DESCRIPTION
**Pull request by maxme**
fix #3327 and #2704

Sorry for `.agignore` it was in the repo, but it should be local.

### CODE REVIEW

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1052,4 +1060,16 @@
     <string name="tabbar_accessibility_label_my_site">My Site</string>
     <string name="tabbar_accessibility_label_me">Me</string>
 
+    <string name="date_range_start_date">Start Date</string>
+    <string name="date_range_end_date">End Date</string>
+
+    <!-- Special characters -->
+    <string name="bullet" translatable="false">\u2022</string>
+    <string name="previous_button" translatable="false">\u2022</string>
```
Comment (maxme) [inline WordPress/src/main/res/values/strings.xml]: ooops

File: WordPress/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -754,6 +761,7 @@
     <string name="add_account_blog_url">Blog address</string>
     <string name="wordpress_blog">WordPress blog</string>
     <string name="blogusername">blogusername</string>
+    <string name="wordpress_com_url" translatable="false">.wordpress.com</string>
```
Comment (nbradbury) [inline WordPress/src/main/res/values/strings.xml]: This is a minor point, but I think the string name should denote that the URL is prefixed with "." (down the road I can see us wanting to show the wordpress.com URL elsewhere).

Comment (maxme) [inline WordPress/src/main/res/values/strings.xml]: Renamed to `dot_wordpress_dot_com_url` in a75f1d3

**Comment by nbradbury** [comment]
Looks good :shipit:

**Pull request merged**